integration/src/main/java/com/aws/greengrass/mqtt/moquette/ClientDeviceAuthorizer.java

```diff
@@ -62,20 +63,38 @@ public boolean checkValid(String clientId, String username, byte[] password) {
                     + "client ID and has a valid AWS IoT client certificate.");
             return false;
         }
-
-        boolean canConnect = canDevicePerform(sessionId, "mqtt:connect", "mqtt:clientId:" + clientId);
+        try {
+            canConnect = canDevicePerform(sessionId, "mqtt:connect", "mqtt:clientId:" + clientId);
+        } catch (AuthorizationException e) {
```

Comment:
not all `AuthorizationException`s are invalid sessions, right? So this log isn't correct. This handling should be more specific to catch only invalid sessions.

Reply:
I'll add in handling for `InvalidSessionExceptions` and `AuthorizationExceptions`. Is there a specific log statement to use for `AuthorizationException` as it used to be "Session ID is invalid" before I refactored the code and added `InvalidSessionException` handling?

Reply:
I'd just say "authorization exception" I guess. It seems like yes invalid session is the only one that will be thrown today, but we can't count on that forever.